src/include/miopen/solver.hpp

```diff
@@ -75,7 +75,13 @@ const int wave_size = 64;
 /// There could be multiple solvers of the same algorithm for a problem config.
 struct SolverBase
 {
-    virtual ~SolverBase() = default;
+    SolverBase() {}
+
+    virtual ~SolverBase()         = default;
+    SolverBase(const SolverBase&) = default;
+    SolverBase(SolverBase&&)      = default;
+    SolverBase& operator=(const SolverBase&) = default;
+    SolverBase& operator=(SolverBase&&) = default;
```

Comment:
```suggestion
    virtual ~SolverBase() = default;
    ...
protected:
    SolverBase() = default;
    SolverBase(const SolverBase&) = default;
```
It is enough to add protected default constructor and protected copy constructor